### PR TITLE
Changed authentication of VM (IoT Edge) to SSH keys

### PR DESCRIPTION
**Pull request by koepalex**
* SSH keys are stored in KeyVault and retrieved during test run
* adopted test helper to use private key for SSH connection

### CODE REVIEW

File: tools/e2etesting/DeployEdge.ps1 [hunk 1]
```diff
@@ -16,8 +18,15 @@ if (!$ResourceGroupName) {
     Write-Error "ResourceGroupName not set."
 }
 
-$edgeVmUsername = "sandboxuser"
-$edgeVmPassword = -join ((65..90) + (97..122) + (33..38) + (40..47) + (48..57)| Get-Random -Count 20 | % {[char]$_})
+if (!$KeysPath) {
```
Comment (markusstuhler) [inline tools/e2etesting/DeployEdge.ps1:21]: Why not use a temp directory (see DeployTestEventProcessor.ps1 lines 158/159)?

File: tools/e2etesting/DeployEdge.ps1 [hunk 2]
```diff
@@ -93,6 +102,18 @@ Set-AzKeyVaultSecret -VaultName $keyVault.VaultName -Name 'iot-edge-device-id' -
 Write-Host "Updating 'os' and '__type__'-Tags in Device Twin..."
 Update-AzIotHubDeviceTwin -ResourceGroupName $ResourceGroupName -IotHubName $iotHub.Name -DeviceId $edgeIdentity.Id -Tag @{ "os" = "Linux"; "__type__" = "iiotedge"} | Out-Null
 
+## Generate SSH keys
+$privateKeyFilePath = [System.IO.Path]::Combine($KeysPath, "id_rsa_iotedge")
```
Comment (markusstuhler) [inline tools/e2etesting/DeployEdge.ps1:106]: Join-Path exists in PowerShell as well. :-)

File: tools/e2etesting/steps/deploytestresources.yml [hunk 3]
```diff
@@ -25,6 +25,7 @@ steps:
     scriptPath: '$(BasePath)\DeployEdge.ps1'
     scriptArguments: >
       -ResourceGroupName "$(ResourceGroupName)"
+      -KeysPath "$(KeysPath)"
```
Comment (markusstuhler) [inline tools/e2etesting/steps/deploytestresources.yml:28]: See above

File: tools/e2etesting/steps/variables.yml [hunk 4]
```diff
@@ -5,4 +5,5 @@ variables:
   AgentPool: Azure-IoT-Manufacturing
   IAILocalFilename: Microsoft.Azure.IIoT.Deployment.exe
   TestPath: $(System.DefaultWorkingDirectory)\e2e-tests\IIoTPlatform-E2E-Tests
-  Runtime: 'win-x64'
\ No newline at end of file
+  Runtime: 'win-x64'
+  KeysPath: $(System.DefaultWorkingDirectory)\keys
```
Comment (markusstuhler) [inline tools/e2etesting/steps/variables.yml:9]: See above.

File: tools/e2etesting/DeployEdge.ps1 (hunk 1 above)
Comment (koepalex) [reply tools/e2etesting/DeployEdge.ps1:21]: WorkingDirectory is cleaned by build agent, this ensure the clean of generated keys.

**Review by markusstuhler — APPROVED** [review]

File: tools/e2etesting/DeployEdge.ps1 (hunk 2 above)
Comment (koepalex) [reply tools/e2etesting/DeployEdge.ps1]: updated

File: tools/e2etesting/DeployEdge.ps1 (hunk 1 above)
Comment (koepalex) [reply tools/e2etesting/DeployEdge.ps1:21]: changed to use $(Agent.TempDirectory) , which is by definition cleaned after each build

File: tools/e2etesting/steps/deploytestresources.yml (hunk 3 above)
Comment (koepalex) [reply tools/e2etesting/steps/deploytestresources.yml]: changed to use $(Agent.TempDirectory) , which is by definition cleaned after each build

File: tools/e2etesting/steps/variables.yml (hunk 4 above)
Comment (koepalex) [reply tools/e2etesting/steps/variables.yml]: changed to use $(Agent.TempDirectory) , which is by definition cleaned after each build

**Pull request merged**